File: elixpo.gitBattle/meme_server.py
#!/usr/bin/env python3
"""
GitHub Battle Report - Python Meme Generator Server
This is a complete example for server-side meme generation using Pillow.

SETUP:
    pip install Pillow Flask Werkzeug

USAGE:
    python meme_server.py
    
Then in your HTML, make POST requests to http://localhost:5000/api/generate-meme

EXAMPLE CLIENT CODE:
    const response = await fetch('http://localhost:5000/api/generate-meme', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
            type: 'drake',
            topText: 'Your Text Here',
            bottomText: 'Another Text'
        })
    });
    const blob = await response.blob();
    const url = URL.createObjectURL(blob);
"""

from flask import Flask, request, send_file
from PIL import Image, ImageDraw, ImageFont
import io
import os
from typing import Dict, Tuple

app = Flask(__name__)

# Configuration
MEME_WIDTH = 600
MEME_HEIGHT = 600
DEFAULT_FONT_SIZE = 28


class MemeGenerator:
    """Generate memes using Pillow"""
    
    @staticmethod
    def wrap_text(text: str, max_width: int, font) -> str:
        """Wrap text to fit within max width"""
        words = text.split()
        lines = []
        current_line = []
        
        for word in words:
            current_line.append(word)
            line_text = ' '.join(current_line)
            bbox = font.getbbox(line_text)
            line_width = bbox[2] - bbox[0]
            
            if line_width > max_width:
                current_line.pop()
                if current_line:
                    lines.append(' '.join(current_line))
                current_line = [word]
        
        if current_line:
            lines.append(' '.join(current_line))
        
        return '\n'.join(lines)
    
    @staticmethod
    def draw_text_with_outline(draw, position: Tuple[int, int], text: str, 
                               font, fill_color: str, outline_color: str):
        """Draw text with outline/stroke effect"""
        x, y = position
        # Draw outline
        for adj_x in [-2, -1, 0, 1, 2]:
            for adj_y in [-2, -1, 0, 1, 2]:
                if adj_x != 0 or adj_y != 0:
                    draw.text((x + adj_x, y + adj_y), text, font=font, fill=outline_color)
        # Draw main text
        draw.text(position, text, font=font, fill=fill_color)
    
    @staticmethod
    def generate_drake(top_text: str, bottom_text: str) -> Image.Image:
        """Drake "Hotline Bling" meme"""
        img = Image.new('RGB', (MEME_WIDTH, MEME_HEIGHT), color='white')
        draw = ImageDraw.Draw(img)
        
        # Get font
        try:
            font = ImageFont.truetype("arial.ttf", DEFAULT_FONT_SIZE)
        except:
            font = ImageFont.load_default()
        
        # Top section (disapprove) - gray
        draw.rectangle([(0, 0), (MEME_WIDTH, MEME_HEIGHT//2)], fill='#E8E8E8')
        
        # Bottom section (approve) - lighter gray
        draw.rectangle([(0, MEME_HEIGHT//2), (MEME_WIDTH, MEME_HEIGHT)], fill='#F5F5F5')
        
        # Wrap and draw text
        top_wrapped = MemeGenerator.wrap_text(top_text, MEME_WIDTH - 100, font)
        bottom_wrapped = MemeGenerator.wrap_text(bottom_text, MEME_WIDTH - 100, font)
        
        MemeGenerator.draw_text_with_outline(
            draw, (MEME_WIDTH//2 - 150, MEME_HEIGHT//4 - 30), 
            top_wrapped, font, '#000000', '#FFFFFF'
        )
        
        MemeGenerator.draw_text_with_outline(
            draw, (MEME_WIDTH//2 - 150, 3*MEME_HEIGHT//4 - 30), 
            bottom_wrapped, font, '#000000', '#FFFFFF'
        )
        
        # Add X and checkmark
        try:
            emoji_font = ImageFont.truetype("arial.ttf", 60)
        except:
            emoji_font = font
        
        draw.text((50, MEME_HEIGHT//4), '❌', font=emoji_font, fill='#FF0000')
        draw.text((50, 3*MEME_HEIGHT//4), '✅', font=emoji_font, fill='#00AA00')
        
        return img
    
    @staticmethod
    def generate_distracted_boyfriend(left_text: str, right_text: str) -> Image.Image:
        """Distracted Boyfriend meme"""
        img = Image.new('RGB', (MEME_WIDTH, MEME_HEIGHT), color='white')
        draw = ImageDraw.Draw(img)
        
        try:
            font = ImageFont.truetype("arial.ttf", 24)
        except:
            font = ImageFont.load_default()
        
        # Left side (ignoring) - pink
        draw.rectangle([(0, 0), (MEME_WIDTH//2, MEME_HEIGHT)], fill='#FFB6D9')
        
        # Right side (looking at) - cyan
        draw.rectangle([(MEME_WIDTH//2, 0), (MEME_WIDTH, MEME_HEIGHT)], fill='#ADD8E6')
        
        # Text
        left_wrapped = MemeGenerator.wrap_text(left_text, MEME_WIDTH//2 - 50, font)
        right_wrapped = MemeGenerator.wrap_text(right_text, MEME_WIDTH//2 - 50, font)
        
        MemeGenerator.draw_text_with_outline(
            draw, (MEME_WIDTH//4 - 100, MEME_HEIGHT//2 - 30), 
            left_wrapped, font, '#000000', '#FFFFFF'
        )
        
        MemeGenerator.draw_text_with_outline(
            draw, (3*MEME_WIDTH//4 - 100, MEME_HEIGHT//2 - 30), 
            right_wrapped, font, '#000000', '#FFFFFF'
        )
        
        # Add emoji labels
        try:
            emoji_font = ImageFont.truetype("arial.ttf", 50)
        except:
            emoji_font = font
        
        draw.text((MEME_WIDTH//4 - 25, 100), '👨', font=emoji_font)
        draw.text((3*MEME_WIDTH//4 - 25, 100), '👩', font=emoji_font)
        
        return img
    
    @staticmethod
    def generate_galaxy_brain(line1: str, line2: str, line3: str, line4: str) -> Image.Image:
        """Galaxy Brain (4 levels)"""
        img = Image.new('RGB', (MEME_WIDTH, MEME_HEIGHT), color='white')
        draw = ImageDraw.Draw(img)
        
        try:
            font = ImageFont.truetype("arial.ttf", 20)
        except:
            font = ImageFont.load_default()
        
        colors = ['#FF6B9D', '#C44569', '#FFA502', '#FFD32A']
        lines = [line1, line2, line3, line4]
        box_height = MEME_HEIGHT // 4
        
        for i, (text, color) in enumerate(zip(lines, colors)):
            y_pos = i * box_height
            
            # Background
            draw.rectangle([(0, y_pos), (MEME_WIDTH, y_pos + box_height)], fill=color)
            
            # Border
            draw.rectangle([(0, y_pos), (MEME_WIDTH, y_pos + box_height)], outline='#000000', width=2)
            
            # Text
            wrapped_text = MemeGenerator.wrap_text(text, MEME_WIDTH - 50, font)
            MemeGenerator.draw_text_with_outline(
                draw, (MEME_WIDTH//2 - 100, y_pos + box_height//2 - 20),
                wrapped_text, font, '#FFFFFF', '#000000'
            )
        
        return img
    
    @staticmethod
    def generate_doge(text1: str, text2: str, text3: str, text4: str) -> Image.Image:
        """Doge meme with Comic Sans style"""
        img = Image.new('RGB', (MEME_WIDTH, MEME_HEIGHT), color='#FFD4A3')
        draw = ImageDraw.Draw(img)
        
        try:
            font = ImageFont.truetype("arial.ttf", 24)
        except:
            font = ImageFont.load_default()
        
        # Text elements
        texts = [
            ('wow', (100, 100), '#FF6B6B'),
            (text1, (350, 80), '#4ECDC4'),
            (text2, (150, 280), '#95E1D3'),
            (text3, (380, 350), '#F7B731'),
            (text4, (300, 450), '#EE5A6F')
        ]
        
        for text, (x, y), color in texts:
            MemeGenerator.draw_text_with_outline(draw, (x, y), text, font, color, '#000000')
        
        # Center circle
        draw.ellipse([(MEME_WIDTH//2 - 80, MEME_HEIGHT//2 - 80), 
                      (MEME_WIDTH//2 + 80, MEME_HEIGHT//2 + 80)], 
                     fill='rgba(255, 255, 255, 128)', outline='#CCCCCC')
        
        # Much text
        much_text = 'much code'
        MemeGenerator.draw_text_with_outline(
            draw, (MEME_WIDTH//2 - 80, MEME_HEIGHT//2 - 15),
            much_text, font, '#FF6B6B', '#000000'
        )
        
        return img
    
    @staticmethod
    def generate_gru_plan(stage1: str, stage2: str, stage3: str) -> Image.Image:
        """Gru's Plan meme"""
        img = Image.new('RGB', (MEME_WIDTH, MEME_HEIGHT), color='#F5DEB3')
        draw = ImageDraw.Draw(img)
        
        try:
            font = ImageFont.truetype("arial.ttf", 14)
        except:
            font = ImageFont.load_default()
        
        box_width = 150
        box_height = 150
        start_y = 100
        
        stages = [stage1, stage2, stage3]
        
        for i, stage_text in enumerate(stages):
            x = 75 + i * 150
            
            # Board
            draw.rectangle([(x, start_y), (x + box_width, start_y + box_height)], 
                          fill='#D2B48C', outline='#8B7355', width=3)
            
            # Checkmark/X/Question mark
            try:
                emoji_font = ImageFont.truetype("arial.ttf", 60)
            except:
                emoji_font = font
            
            if i == 2:
                draw.text((x + box_width//2 - 20, start_y + 30), '✗', 
                         font=emoji_font, fill='#FF0000')
            elif i == 1:
                draw.text((x + box_width//2 - 20, start_y + 30), '?',
                         font=emoji_font, fill='#FFD700')
            else:
                draw.text((x + box_width//2 - 20, start_y + 30), '✓',
                         font=emoji_font, fill='#00AA00')
            
            # Text below
            MemeGenerator.draw_text_with_outline(
                draw, (x + box_width//2 - 50, start_y + box_height + 20),
                stage_text, font, '#000000', '#FFFFFF'
            )
        
        # Gru's face
        draw.ellipse([(10, 280), (90, 360)], fill='#FFD4A3', outline='#000000', width=2)
        draw.text((40, 310), '😈', font=emoji_font if 'emoji_font' in locals() else font)
        
        return img


@app.route('/api/generate-meme', methods=['POST'])
def generate_meme():
    """Generate meme endpoint"""
    try:
        data = request.json
        meme_type = data.get('type', 'drake')
        
        # Route to appropriate generator
        if meme_type == 'drake':
            img = MemeGenerator.generate_drake(
                data.get('topText', 'Top Text'),
                data.get('bottomText', 'Bottom Text')
            )
        elif meme_type == 'distracted':
            img = MemeGenerator.generate_distracted_boyfriend(
                data.get('leftText', 'Left Text'),
                data.get('rightText', 'Right Text')
            )
        elif meme_type == 'galaxy-brain':
            img = MemeGenerator.generate_galaxy_brain(
                data.get('line1', 'Level 1'),
                data.get('line2', 'Level 2'),
                data.get('line3', 'Level 3'),
                data.get('line4', 'Level 4')
            )
        elif meme_type == 'doge':
            img = MemeGenerator.generate_doge(
                data.get('text1', 'Text 1'),
                data.get('text2', 'Text 2'),
                data.get('text3', 'Text 3'),
                data.get('text4', 'Text 4')
            )
        elif meme_type == 'gru':
            img = MemeGenerator.generate_gru_plan(
                data.get('stage1', 'Stage 1'),
                data.get('stage2', 'Stage 2'),
                data.get('stage3', 'Stage 3')
            )
        else:
            return {'error': 'Unknown meme type'}, 400
        
        # Convert to bytes
        img_io = io.BytesIO()
        img.save(img_io, 'PNG', quality=95)
        img_io.seek(0)
        
        return send_file(img_io, mimetype='image/png')
    
    except Exception as e:
        return {'error': str(e)}, 500


@app.route('/api/health', methods=['GET'])
def health():
    """Health check endpoint"""
    return {'status': 'ok', 'service': 'GitHub Battle Meme Generator'}


if __name__ == '__main__':
    print("🚀 Starting GitHub Battle Meme Generator Server...")
    print("📍 API running at http://localhost:5000")
    print("🎨 POST to http://localhost:5000/api/generate-meme with meme config")
    print("✅ Check health at http://localhost:5000/api/health")
    app.run(debug=True, port=5000)

File: elixpo.gitBattle/meme_server.js
/**
 * GitHub Battle Report - Node.js Meme Generator Server
 * This is a complete example for server-side meme generation using Canvas.
 * 
 * SETUP:
 *     npm init -y
 *     npm install express canvas cors body-parser
 * 
 * USAGE:
 *     node meme_server.js
 * 
 * Then in your HTML, make POST requests to http://localhost:5000/api/generate-meme
 * 
 * EXAMPLE CLIENT CODE:
 *     const response = await fetch('http://localhost:5000/api/generate-meme', {
 *         method: 'POST',
 *         headers: { 'Content-Type': 'application/json' },
 *         body: JSON.stringify({
 *             type: 'drake',
 *             topText: 'Your Text Here',
 *             bottomText: 'Another Text'
 *         })
 *     });
 *     const blob = await response.blob();
 *     const url = URL.createObjectURL(blob);
 */

const express = require('express');
const { createCanvas } = require('canvas');
const cors = require('cors');
const bodyParser = require('body-parser');
const fs = require('fs');

const app = express();

// Middleware
app.use(cors());
app.use(bodyParser.json({ limit: '10mb' }));

// Constants
const MEME_WIDTH = 600;
const MEME_HEIGHT = 600;
const DEFAULT_FONT_SIZE = 28;

class MemeGenerator {
    /**
     * Wrap text to fit within max width
     */
    static wrapText(ctx, text, maxWidth, fontSize) {
        const words = text.split(' ');
        const lines = [];
        let currentLine = '';

        words.forEach(word => {
            const testLine = currentLine + (currentLine ? ' ' : '') + word;
            const metrics = ctx.measureText(testLine);

            if (metrics.width > maxWidth && currentLine) {
                lines.push(currentLine);
                currentLine = word;
            } else {
                currentLine = testLine;
            }
        });

        if (currentLine) lines.push(currentLine);
        return lines;
    }

    /**
     * Draw text with outline/stroke effect
     */
    static drawTextWithOutline(ctx, text, x, y, fillColor, outlineColor, fontSize) {
        ctx.font = `bold ${fontSize}px Arial`;
        ctx.strokeStyle = outlineColor;
        ctx.lineWidth = 3;
        ctx.fillStyle = fillColor;

        // Draw outline
        for (let adj_x = -2; adj_x <= 2; adj_x++) {
            for (let adj_y = -2; adj_y <= 2; adj_y++) {
                if (adj_x !== 0 || adj_y !== 0) {
                    ctx.strokeText(text, x + adj_x, y + adj_y);
                }
            }
        }

        // Draw main text
        ctx.fillText(text, x, y);
    }

    /**
     * Generate Drake "Hotline Bling" meme
     */
    static generateDrake(topText, bottomText) {
        const canvas = createCanvas(MEME_WIDTH, MEME_HEIGHT);
        const ctx = canvas.getContext('2d');

        // Top section (disapprove) - gray
        ctx.fillStyle = '#E8E8E8';
        ctx.fillRect(0, 0, MEME_WIDTH, MEME_HEIGHT / 2);

        // Bottom section (approve) - lighter gray
        ctx.fillStyle = '#F5F5F5';
        ctx.fillRect(0, MEME_HEIGHT / 2, MEME_WIDTH, MEME_HEIGHT / 2);

        // Set font
        ctx.font = `bold ${DEFAULT_FONT_SIZE}px Arial`;
        ctx.textAlign = 'center';

        // Wrap and draw text
        const topLines = this.wrapText(ctx, topText, MEME_WIDTH - 100, DEFAULT_FONT_SIZE);
        const bottomLines = this.wrapText(ctx, bottomText, MEME_WIDTH - 100, DEFAULT_FONT_SIZE);

        let topY = MEME_HEIGHT / 4 - (topLines.length * DEFAULT_FONT_SIZE) / 2;
        topLines.forEach(line => {
            this.drawTextWithOutline(ctx, line, MEME_WIDTH / 2, topY, '#000000', '#FFFFFF', DEFAULT_FONT_SIZE);
            topY += DEFAULT_FONT_SIZE;
        });

        let bottomY = (3 * MEME_HEIGHT) / 4 - (bottomLines.length * DEFAULT_FONT_SIZE) / 2;
        bottomLines.forEach(line => {
            this.drawTextWithOutline(ctx, line, MEME_WIDTH / 2, bottomY, '#000000', '#FFFFFF', DEFAULT_FONT_SIZE);
            bottomY += DEFAULT_FONT_SIZE;
        });

        // Add X and checkmark
        ctx.font = 'bold 60px Arial';
        ctx.fillStyle = '#FF0000';
        ctx.fillText('❌', 50, MEME_HEIGHT / 4);
        ctx.fillStyle = '#00AA00';
        ctx.fillText('✅', 50, (3 * MEME_HEIGHT) / 4);

        return canvas;
    }

    /**
     * Generate Distracted Boyfriend meme
     */
    static generateDistractionMeme(leftText, rightText) {
        const canvas = createCanvas(MEME_WIDTH, MEME_HEIGHT);
        const ctx = canvas.getContext('2d');

        // Left side (ignoring) - pink
        ctx.fillStyle = '#FFB6D9';
        ctx.fillRect(0, 0, MEME_WIDTH / 2, MEME_HEIGHT);

        // Right side (looking at) - cyan
        ctx.fillStyle = '#ADD8E6';
        ctx.fillRect(MEME_WIDTH / 2, 0, MEME_WIDTH / 2, MEME_HEIGHT);

        // Text
        ctx.font = `bold 24px Arial`;
        ctx.textAlign = 'center';

        const leftLines = this.wrapText(ctx, leftText, MEME_WIDTH / 2 - 50, 24);
        const rightLines = this.wrapText(ctx, rightText, MEME_WIDTH / 2 - 50, 24);

        let leftY = MEME_HEIGHT / 2 - (leftLines.length * 24) / 2;
        leftLines.forEach(line => {
            this.drawTextWithOutline(ctx, line, MEME_WIDTH / 4, leftY, '#000000', '#FFFFFF', 24);
            leftY += 24;
        });

        let rightY = MEME_HEIGHT / 2 - (rightLines.length * 24) / 2;
        rightLines.forEach(line => {
            this.drawTextWithOutline(ctx, line, (3 * MEME_WIDTH) / 4, rightY, '#000000', '#FFFFFF', 24);
            rightY += 24;
        });

        // Add emoji labels
        ctx.font = 'bold 50px Arial';
        ctx.fillStyle = '#000';
        ctx.fillText('👨', MEME_WIDTH / 4 - 25, 100);
        ctx.fillText('👩', (3 * MEME_WIDTH) / 4 - 25, 100);

        return canvas;
    }

    /**
     * Generate Galaxy Brain meme (4 levels)
     */
    static generateGalaxyBrain(line1, line2, line3, line4) {
        const canvas = createCanvas(MEME_WIDTH, MEME_HEIGHT);
        const ctx = canvas.getContext('2d');

        const colors = ['#FF6B9D', '#C44569', '#FFA502', '#FFD32A'];
        const lines = [line1, line2, line3, line4];
        const boxHeight = MEME_HEIGHT / 4;

        ctx.font = 'bold 20px Arial';
        ctx.textAlign = 'center';

        lines.forEach((text, i) => {
            const y = i * boxHeight;

            // Background
            ctx.fillStyle = colors[i];
            ctx.fillRect(0, y, MEME_WIDTH, boxHeight);

            // Border
            ctx.strokeStyle = '#000000';
            ctx.lineWidth = 3;
            ctx.strokeRect(0, y, MEME_WIDTH, boxHeight);

            // Text
            const wrapped = this.wrapText(ctx, text, MEME_WIDTH - 50, 20);
            let textY = y + boxHeight / 2 - (wrapped.length * 20) / 2;
            wrapped.forEach(line => {
                this.drawTextWithOutline(ctx, line, MEME_WIDTH / 2, textY, '#FFFFFF', '#000000', 20);
                textY += 20;
            });
        });

        return canvas;
    }

    /**
     * Generate Doge meme
     */
    static generateDoge(text1, text2, text3, text4) {
        const canvas = createCanvas(MEME_WIDTH, MEME_HEIGHT);
        const ctx = canvas.getContext('2d');

        // Background
        ctx.fillStyle = '#FFD4A3';
        ctx.fillRect(0, 0, MEME_WIDTH, MEME_HEIGHT);

        // Text elements
        const textElements = [
            { text: 'wow', x: 100, y: 100, color: '#FF6B6B', size: 32 },
            { text: text1, x: 350, y: 80, color: '#4ECDC4', size: 24 },
            { text: text2, x: 150, y: 280, color: '#95E1D3', size: 24 },
            { text: text3, x: 380, y: 350, color: '#F7B731', size: 24 },
            { text: text4, x: 300, y: 450, color: '#EE5A6F', size: 24 }
        ];

        textElements.forEach(elem => {
            this.drawTextWithOutline(ctx, elem.text, elem.x, elem.y, elem.color, '#000000', elem.size);
        });

        // Center circle
        ctx.fillStyle = 'rgba(255, 255, 255, 0.3)';
        ctx.beginPath();
        ctx.arc(MEME_WIDTH / 2, MEME_HEIGHT / 2, 80, 0, Math.PI * 2);
        ctx.fill();

        // "Much" text
        ctx.font = 'bold 32px Arial';
        ctx.fillStyle = '#FF6B6B';
        ctx.textAlign = 'center';
        ctx.fillText('much code', MEME_WIDTH / 2, MEME_HEIGHT / 2 + 10);

        return canvas;
    }

    /**
     * Generate Gru's Plan meme
     */
    static generateGruPlan(stage1, stage2, stage3) {
        const canvas = createCanvas(MEME_WIDTH, MEME_HEIGHT);
        const ctx = canvas.getContext('2d');

        // Background
        ctx.fillStyle = '#F5DEB3';
        ctx.fillRect(0, 0, MEME_WIDTH, MEME_HEIGHT);

        const boxWidth = 150;
        const boxHeight = 150;
        const startY = 100;
        const stages = [stage1, stage2, stage3];

        stages.forEach((stageText, i) => {
            const x = 75 + i * 150;

            // Board
            ctx.fillStyle = '#D2B48C';
            ctx.fillRect(x, startY, boxWidth, boxHeight);

            // Border
            ctx.strokeStyle = '#8B7355';
            ctx.lineWidth = 3;
            ctx.strokeRect(x, startY, boxWidth, boxHeight);

            // Checkmark/X/Question
            ctx.font = 'bold 60px Arial';
            ctx.textAlign = 'center';
            if (i === 2) {
                ctx.fillStyle = '#FF0000';
                ctx.fillText('✗', x + boxWidth / 2, startY + 50);
            } else if (i === 1) {
                ctx.fillStyle = '#FFD700';
                ctx.fillText('?', x + boxWidth / 2, startY + 50);
            } else {
                ctx.fillStyle = '#00AA00';
                ctx.fillText('✓', x + boxWidth / 2, startY + 50);
            }

            // Text below
            ctx.font = 'bold 14px Arial';
            const wrapped = this.wrapText(ctx, stageText, boxWidth - 20, 14);
            let textY = startY + boxHeight + 20;
            wrapped.forEach(line => {
                this.drawTextWithOutline(ctx, line, x + boxWidth / 2, textY, '#000000', '#FFFFFF', 14);
                textY += 14;
            });
        });

        // Gru's face
        ctx.fillStyle = '#FFD4A3';
        ctx.beginPath();
        ctx.arc(50, 320, 40, 0, Math.PI * 2);
        ctx.fill();
        ctx.strokeStyle = '#000000';
        ctx.lineWidth = 2;
        ctx.stroke();

        ctx.font = 'bold 30px Arial';
        ctx.fillText('😈', 50, 330);

        return canvas;
    }
}

/**
 * POST /api/generate-meme
 * Generate meme based on type and parameters
 */
app.post('/api/generate-meme', (req, res) => {
    try {
        const { type, topText, bottomText, leftText, rightText, line1, line2, line3, line4, text1, text2, text3, text4, stage1, stage2, stage3 } = req.body;

        let canvas;

        switch (type) {
            case 'drake':
                canvas = MemeGenerator.generateDrake(topText || 'Top Text', bottomText || 'Bottom Text');
                break;
            case 'distracted':
                canvas = MemeGenerator.generateDistractionMeme(leftText || 'Left', rightText || 'Right');
                break;
            case 'galaxy-brain':
                canvas = MemeGenerator.generateGalaxyBrain(line1 || 'Level 1', line2 || 'Level 2', line3 || 'Level 3', line4 || 'Level 4');
                break;
            case 'doge':
                canvas = MemeGenerator.generateDoge(text1 || 'Text 1', text2 || 'Text 2', text3 || 'Text 3', text4 || 'Text 4');
                break;
            case 'gru':
                canvas = MemeGenerator.generateGruPlan(stage1 || 'Stage 1', stage2 || 'Stage 2', stage3 || 'Stage 3');
                break;
            default:
                return res.status(400).json({ error: 'Unknown meme type' });
        }

        // Convert canvas to PNG buffer
        const buffer = canvas.toBuffer('image/png');
        res.type('image/png');
        res.send(buffer);
    } catch (error) {
        console.error('Error generating meme:', error);
        res.status(500).json({ error: error.message });
    }
});

/**
 * GET /api/health
 * Health check endpoint
 */
app.get('/api/health', (req, res) => {
    res.json({ status: 'ok', service: 'GitHub Battle Meme Generator' });
});

/**
 * Start server
 */
const PORT = process.env.PORT || 5000;
app.listen(PORT, () => {
    console.log('🚀 Starting GitHub Battle Meme Generator Server...');
    console.log(`📍 API running at http://localhost:${PORT}`);
    console.log(`🎨 POST to http://localhost:${PORT}/api/generate-meme with meme config`);
    console.log(`✅ Check health at http://localhost:${PORT}/api/health`);
});
